src/routes/index.js

```diff
@@ -46,13 +48,13 @@ router.post('/api/users/passwordreset/:token', decodeResetPasswordToken, checkEm
 router.post('/api/users/logout', verifyToken, UserController.signOut);
 
 router.get('/api/user/:username', usernameCheck, UserProfile.getProfile);
-router.patch('/api/users/:username', verifyToken, bodyValidation, usernameAvailability, UserProfile.updateProfile);
 router.get('/api/allusers/', verifyToken, UserProfile.getAllUser);
 router.post('/api/users/login', signinValidation, AuthController.signin);
 router.patch('/api/users/verification/:userToken', UserController.verifyUser);
 
-router.patch('/api/users/:username', verifyToken, bodyValidation, usernameAvailability, UserProfile.updateProfile);
-router.get('/api/users/', verifyToken, UserProfile.getAllUser);
+router.patch('/api/users', verifyToken, UserProfile.updateProfile);
+router.get('/api/users', verifyToken, UserProfile.getAllUser);
 
+router.delete('/api/users/:id', verifyToken, checkAction('User'), UserController.deleteUser);
```

Comment:
Why are you having this action `checkAction('User')` here and not at the controller level?

Reply:
`checkAction` is a middleware function that checks if a user trying to access the router has the priority of deleting in a user table.